hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/query/AbstractRelationQueryGenerator.java

```diff
@@ -65,7 +67,11 @@ public Query getQuery(SharedSessionContractImplementor session, Object primaryKe
 		final String queryString = getQueryString( session.getFactory(), removed );
 
 		final Query query = session.createQuery( queryString );
-		query.setParameter( DEL_REVISION_TYPE_PARAMETER, RevisionType.DEL );
+		query.setParameter(
+				DEL_REVISION_TYPE_PARAMETER,
+				RevisionType.DEL,
+				new CustomType( new RevisionTypeType(), session.getFactory().getTypeConfiguration() )
```

Comment:
Wouldn't `sessionFactory.getTypeConfiguration().getBasicTypeRegistry().getRegisteredType( RevisionTypeType.class.getName() )` also work instead of creating a new type all the time? The same would apply to `org.hibernate.envers.internal.tools.query.QueryBuilder#toQuery`

Reply:
I'll give it a look, I simply reused what was being used in QueryBuilder that had been changed for 6.0 earlier.

Reply:
Worked like a charm, thanks @beikov !